src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,93 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int CAPACITY = 10;
+    private static final double COEFFICIENT = 0.5;
+    private Object[] array = new Object[CAPACITY];
+    private int size = 0;
 
     @Override
     public void add(T value) {
-
+        if (size < array.length) {
```

Comment:
let's move resizing logic in a separate private method

if (size is equal to array length) -> resize
after we resized or not, we add the value